osbenchmark/telemetry.py

```diff
@@ -409,48 +415,49 @@ def record(self):
-        except elasticsearch.TransportError:
-            msg = "A transport error occurred while collecting CCR stats from the endpoint [{}?filter_path={}] on " \
-                  "cluster [{}]".format(ccr_stats_api_endpoint, filter_path, self.cluster_name)
-            self.logger.exception(msg)
-            raise exceptions.BenchmarkError(msg)
+        for index in self.indices:
+            try:
+                stats = self.client.transport.perform_request("GET", "/_plugins/_replication/" + index + "/_status")
+                if stats["status"] == "SYNCING":
+                    self.record_stats_per_index(index, stats)
+                else:
+                    self.logger.info("CCR Status is not syncing. Ignoring for now!")
+            except elasticsearch.TransportError:
+                msg = "A transport error occurred while collecting CCR stats for remote cluster: {}".format(self.cluster_name)
+                self.logger.exception(msg)
+                raise exceptions.BenchmarkError(msg)
 
-        if filter_path in stats and "indices" in stats[filter_path]:
-            for indices in stats[filter_path]["indices"]:
-                try:
-                    if self.indices and indices["index"] not in self.indices:
-                        # Skip metrics for indices not part of user supplied whitelist (ccr-stats-indices) in telemetry params.
-                        continue
-                    self.record_stats_per_index(indices["index"], indices["shards"])
-                except KeyError:
-                    self.logger.warning(
-                        "The 'indices' key in %s does not contain an 'index' or 'shards' key "
-                        "Maybe the output format of the %s endpoint has changed. Skipping.", ccr_stats_api_endpoint, ccr_stats_api_endpoint
-                    )
 
     def record_stats_per_index(self, name, stats):
         """
         :param name: The index name.
         :param stats: A dict with returned CCR stats for the index.
         """
+        doc = {
+            "name": "ccr-status",
+            "index": name,
+            "leader_checkpoint": stats["syncing_details"]["leader_checkpoint"],
+            "follower_checkpoint": stats["syncing_details"]["follower_checkpoint"],
+            "replication_lag": self.calculate_lag(name, stats["syncing_details"]["leader_checkpoint"],
+                                    stats["syncing_details"]["follower_checkpoint"])
+        }
+        index_metadata = {
+            "cluster": self.cluster_name,
+            "index": name
+        }
+        self.metrics_store.put_doc(doc, level=MetaInfoScope.cluster, meta_data=index_metadata)
+
+    def calculate_lag(self, index, leader_checkpoint, follower_checkpoint):
+        leader_checkpoint_queue = self.leader_checkpoints[index]
+
+        while(len(leader_checkpoint_queue) > 0 and leader_checkpoint_queue[0] <= follower_checkpoint):
```

Comment:
Nit: I think you can remove the `len` call and change this to `while(leader_checkpoint_queue and leader_checkpoint_queue[0] <= follower_checkpoint):`

Reply:
Sure.